From patchwork Fri Sep 28 13:33:21 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Tony Krowiak <akrowiak@linux.vnet.ibm.com>
X-Patchwork-Id: 10619945
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 8BC03174A
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 28 Sep 2018 13:33:39 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 7A90E2B30A
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 28 Sep 2018 13:33:39 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id 6E02E2B31E; Fri, 28 Sep 2018 13:33:39 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-7.9 required=2.0 tests=BAYES_00,MAILING_LIST_MULTI,
	RCVD_IN_DNSWL_HI autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 0D56D2B30A
	for <patchwork-kvm@patchwork.kernel.org>;
 Fri, 28 Sep 2018 13:33:38 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1729183AbeI1T5Y (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Fri, 28 Sep 2018 15:57:24 -0400
Received: from mx0b-001b2d01.pphosted.com ([148.163.158.5]:58354 "EHLO
        mx0a-001b2d01.pphosted.com" rhost-flags-OK-OK-OK-FAIL)
        by vger.kernel.org with ESMTP id S1728953AbeI1T5X (ORCPT
        <rfc822;kvm@vger.kernel.org>); Fri, 28 Sep 2018 15:57:23 -0400
Received: from pps.filterd (m0098421.ppops.net [127.0.0.1])
        by mx0a-001b2d01.pphosted.com (8.16.0.22/8.16.0.22) with SMTP id
 w8SDODNB071191
        for <kvm@vger.kernel.org>; Fri, 28 Sep 2018 09:33:34 -0400
Received: from e32.co.us.ibm.com (e32.co.us.ibm.com [32.97.110.150])
        by mx0a-001b2d01.pphosted.com with ESMTP id 2mskfgdhhj-1
        (version=TLSv1.2 cipher=AES256-GCM-SHA384 bits=256 verify=NOT)
        for <kvm@vger.kernel.org>; Fri, 28 Sep 2018 09:33:34 -0400
Received: from localhost
        by e32.co.us.ibm.com with IBM ESMTP SMTP Gateway: Authorized Use Only!
 Violators will be prosecuted
        for <kvm@vger.kernel.org> from <akrowiak@linux.vnet.ibm.com>;
        Fri, 28 Sep 2018 07:33:33 -0600
Received: from b03cxnp08025.gho.boulder.ibm.com (9.17.130.17)
        by e32.co.us.ibm.com (192.168.1.132) with IBM ESMTP SMTP Gateway:
 Authorized Use Only! Violators will be prosecuted;
        (version=TLSv1/SSLv3 cipher=AES256-GCM-SHA384 bits=256/256)
        Fri, 28 Sep 2018 07:33:29 -0600
Received: from b03ledav001.gho.boulder.ibm.com
 (b03ledav001.gho.boulder.ibm.com [9.17.130.232])
        by b03cxnp08025.gho.boulder.ibm.com (8.14.9/8.14.9/NCO v10.0) with
 ESMTP id w8SDXQec21364970
        (version=TLSv1/SSLv3 cipher=DHE-RSA-AES256-GCM-SHA384 bits=256
 verify=FAIL);
        Fri, 28 Sep 2018 06:33:27 -0700
Received: from b03ledav001.gho.boulder.ibm.com (unknown [127.0.0.1])
        by IMSVA (Postfix) with ESMTP id DE88D6E052;
        Fri, 28 Sep 2018 07:33:26 -0600 (MDT)
Received: from b03ledav001.gho.boulder.ibm.com (unknown [127.0.0.1])
        by IMSVA (Postfix) with ESMTP id CBBC96E05B;
        Fri, 28 Sep 2018 07:33:24 -0600 (MDT)
Received: from oc8043147753.ibm.com.endicott.ibm.com (unknown [9.60.75.213])
        by b03ledav001.gho.boulder.ibm.com (Postfix) with ESMTPS;
        Fri, 28 Sep 2018 07:33:24 -0600 (MDT)
From: Tony Krowiak <akrowiak@linux.vnet.ibm.com>
To: linux-s390@vger.kernel.org, linux-kernel@vger.kernel.org,
        kvm@vger.kernel.org
Cc: freude@de.ibm.com, schwidefsky@de.ibm.com,
        heiko.carstens@de.ibm.com, borntraeger@de.ibm.com,
        cohuck@redhat.com, kwankhede@nvidia.com,
        bjsdjshi@linux.vnet.ibm.com, pbonzini@redhat.com,
        alex.williamson@redhat.com, pmorel@linux.vnet.ibm.com,
        alifm@linux.vnet.ibm.com, mjrosato@linux.vnet.ibm.com,
        jjherne@linux.vnet.ibm.com, thuth@redhat.com,
        pasic@linux.vnet.ibm.com, fiuczy@linux.vnet.ibm.com,
        buendgen@de.ibm.com, Tony Krowiak <akrowiak@linux.ibm.com>
Subject: [FIXUP v11] fixup! s390: vfio-ap: implement mediated device open
 callback
Date: Fri, 28 Sep 2018 09:33:21 -0400
X-Mailer: git-send-email 2.19.0.221.g150f307
In-Reply-To: <20180928121429.7323d78b.cohuck@redhat.com>
References: <20180928121429.7323d78b.cohuck@redhat.com>
MIME-Version: 1.0
X-TM-AS-GCONF: 00
x-cbid: 18092813-0004-0000-0000-0000149389F5
X-IBM-SpamModules-Scores: 
X-IBM-SpamModules-Versions: BY=3.00009785; HX=3.00000242; KW=3.00000007;
 PH=3.00000004; SC=3.00000267; SDB=6.01094887; UDB=6.00566013; IPR=6.00874923;
 MB=3.00023542; MTD=3.00000008; XFM=3.00000015; UTC=2018-09-28 13:33:32
X-IBM-AV-DETECTION: SAVI=unused REMOTE=unused XFE=unused
x-cbparentid: 18092813-0005-0000-0000-000088F8201B
Message-Id: <20180928133321.31884-1-akrowiak@linux.vnet.ibm.com>
X-Proofpoint-Virus-Version: vendor=fsecure engine=2.50.10434:,,
 definitions=2018-09-28_06:,,
 signatures=0
X-Proofpoint-Spam-Details: rule=outbound_notspam policy=outbound score=0
 priorityscore=1501
 malwarescore=0 suspectscore=0 phishscore=0 bulkscore=0 spamscore=0
 clxscore=1015 lowpriorityscore=0 mlxscore=0 impostorscore=0
 mlxlogscore=818 adultscore=0 classifier=spam adjust=0 reason=mlx
 scancount=1 engine=8.0.1-1807170000 definitions=main-1809280136
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

From: Tony Krowiak <akrowiak@linux.ibm.com>

Fixes case statement in vfio_ap_mdev_copy_masks() function in
vfio-ap-ops.c.
---
 drivers/s390/crypto/vfio_ap_ops.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/s390/crypto/vfio_ap_ops.c b/drivers/s390/crypto/vfio_ap_ops.c
index e1e1beaaeba5..86b42487a51c 100644
--- a/drivers/s390/crypto/vfio_ap_ops.c
+++ b/drivers/s390/crypto/vfio_ap_ops.c
@@ -741,11 +741,12 @@ static void vfio_ap_mdev_copy_masks(struct ap_matrix_mdev *matrix_mdev)
 		break;
 	case CRYCB_FORMAT1:
 	case CRYCB_FORMAT0:
-	default:
 		apm = (unsigned long *)crycb->apcb0.apm;
 		aqm = (unsigned long *)crycb->apcb0.aqm;
 		adm = (unsigned long *)crycb->apcb0.adm;
 		break;
+	default:
+		return;
 	}
 
 	nbytes = DIV_ROUND_UP(matrix_mdev->matrix.apm_max + 1, BITS_PER_BYTE);
